activerecord/lib/active_record/persistence.rb

```diff
@@ -121,7 +121,7 @@ def insert(attributes, returning: nil, unique_by: nil)
       #     { id: 1, title: "Eloquent Ruby", author: "Russ" }
       #   ])
       def insert_all(attributes, returning: nil, unique_by: nil)
-        InsertAll.new(self, attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by).execute
+        InsertAll.new(scope_for_association(current_scope || relation), attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by).execute
```

Comment:
We need to pass `current_scope` and fallback to `relation` (default argument) for now. `scope_for_association` with default argument doesn't preserve `scope_for_create`. Isn't that actually bug? I would expect `scope_for_association` to keep `scope_for_create`. I can change `scope_for_association` to keep that.